src/flash/flashCmd.ts

```diff
@@ -88,33 +104,128 @@ export async function verifyCanFlash(
     if (!listDfu) {
       const errStr = "No DFU capable USB device available found";
       OutputChannel.show();
       OutputChannel.appendLineAndShow(errStr, "Flash");
-      return Logger.errorNotify(
-        errStr,
-        new Error("NO_DFU_DEVICES_FOUND")
-      );
+      return Logger.errorNotify(errStr, new Error("NO_DFU_DEVICES_FOUND"));
     }
   }
   return continueFlag;
 }
+
+export function isFlashEncryptionEnabled(workspaceRoot: vscode.Uri) {
+  const flashEncryption = utils.getConfigValueFromSDKConfig(
+    "CONFIG_FLASH_ENCRYPTION_ENABLED",
+    workspaceRoot
+  );
+  return flashEncryption === "y";
+}
+
+export async function checkFlashEncryption(
+  flashType: ESP.FlashType,
+  workspaceRoot: vscode.Uri
+): Promise<FlashCheckResult> {
+  Logger.info(`Using flash type: ${flashType}`, { tag: "Flash" });
+
+  try {
+    if (flashType !== ESP.FlashType.UART) {
+      const errorMessage = `Invalid flash type for partition encryption. Required: UART, Found: ${flashType}. \n Choose one of the actions presented in the top center quick pick menu and re-flash.`;
+      const error = new Error(errorMessage);
+      const customButtons = [
+        {
+          label: "Change flash type to UART",
+          action: () => {
+            idfConf.writeParameter(
+              "idf.flashType",
+              "UART",
+              vscode.ConfigurationTarget.WorkspaceFolder,
+              workspaceRoot
+            );
+            const saveMessage = locDic.localize(
+              "flash.saveFlashTypeUART",
+              "Flashing method successfully changed to UART"
+            );
+            Logger.infoNotify(saveMessage);
+            OutputChannel.appendLineAndShow(saveMessage, "Flash Encryption");
+          },
+        },
+        {
+          label: "Disable Flash Encryption",
+          action: () => {
+            disableFlashEncryption();
+            const saveMessage = locDic.localize(
+              "menuconfig.disabledFlashEncryption",
+              "Flash encryption has been disabled in the SDK configuration"
+            );
+            Logger.infoNotify(saveMessage);
+            OutputChannel.appendLineAndShow(saveMessage, "Flash Encryption");
+          },
+        },
+      ];
+
+      OutputChannel.appendLineAndShow(errorMessage, "Flash Encryption");
+      Logger.errorNotify(errorMessage, error, { tag: "Flash Encryption" });
+      await showQuickPickWithCustomActions(
+        "Pick one of the following actions to continue",
+        customButtons
+      );
+      return {
+        success: false,
+        resultType: FlashCheckResultType.ErrorInvalidFlashType,
+      };
+    }
+    const eFuse = new ESPEFuseManager(workspaceRoot);
+    const summaryResult = await eFuse.readSummary();
+    if (summaryResult && summaryResult.BLOCK_KEY0) {
+      if (
+        // eFuse is not set
+        summaryResult.BLOCK_KEY0.value &&
+        summaryResult.BLOCK_KEY0.value === ESP.DEFAULT_UNSET_EFUSE_VALUE
```

Comment:
The reference block you use here is not available for esp32 on ESP-IDF v5.0.4.

In the eFuse summary we receive:

```ts
BLOCK1: {
  bit_len: 256,
  block: 1,
  category: "security",
  description: "Flash encryption key",
  efuse_type: "bytes:32",
  name: "BLOCK1",
  pos: 0,
  readable: true,
  value: "00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00",
  word: 0,
  writeable: true,
}
```

This causes the flash to execute as encrypted and fail because of missing eFuse:

```
Executing task: /Users/brainignacio/.espressif/python_env/idf5.0_py3.11_env/bin/python /Users/brainignacio/esp/esp504/esp-idf/components/esptool_py/esptool/esptool.py -p /dev/cu.usbserial-0001 -b 460800 --before default_reset --after no_reset --chip esp32 write_flash --flash_mode dio --flash_freq 40m --flash_size 2MB --encrypt 0x1000 bootloader/bootloader.bin 0x20000 flash_encryption.bin 0x10000 partition_table/partition-table.bin 

esptool.py v4.7.0
Serial port /dev/cu.usbserial-0001
Connecting......
Chip is ESP32-D0WD (revision v0.0)
Features: WiFi, BT, Dual Core, Coding Scheme None
Crystal is 40MHz
MAC: 30:ae:a4:00:45:cc
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 460800
Changed.
Configuring flash size...
Unexpected FLASH_CRYPT_CONFIG value: 0x0
Flash encryption key is not programmed

A fatal error occurred: Can't perform encrypted flash write, consult Flash Encryption documentation for more information
```

Reply:
I've updated this part to search for the property:  "FLASH_CRYPT_CNT" after running espefuse summary command and "SPI_BOOT_CRYPT_CNT" for all the other boards